packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -732,7 +732,7 @@ export const KeyboardNavigationMixin = (superClass) =>
       const cell = e.composedPath()[0];
       if (cell._content && cell._content.firstElementChild) {
         const wasNavigating = this.hasAttribute('navigating');
-        cell._content.firstElementChild.click();
+        cell._content.firstElementChild.dispatchEvent(new MouseEvent('click', { shiftKey: e.shiftKey }));
```

Comment:
Calling `.click()` fires a `click` event that has `bubbles: true`, `cancelable: true` and `composed: true`.
Tested with the following code snippet and Chrome, Firefox and Safari behave the same in this regard:

```html
<button>Click me</button>

<script type="module">
  const button = document.querySelector('button');

  button.addEventListener('click', (e) => {
    console.log(e);
  });

  button.click();

  button.dispatchEvent(new MouseEvent('click'));
</script>
```

Consider changing this fake event to also set `bubbles`, `composed` and `cancelable`.
In theory someone might listen on `click` event on the grid host and expect it to work.

Reply:
Good catch, I had tried to search for what exactly the click() method fired but couldn't find it documented anywhere... don't know why I didn't just test like this

Reply:
Done.